packages/migrate/src/interfaces/plugin.ts

```diff
@@ -23,6 +23,6 @@ export class Plugin {
   }
 
   async run(params: PluginParams<PluginOptions>): PluginResult {
-    return this.service.getFileText(params.fileName);
+    return [params.fileName];
```

Comment:
Confirming this no longer needs to be passed through the service?

Reply:
Plugin input: changed file from transformers, including file name and new text
What Plugin does: sets the new text with the service, and iterates the changed text through diagnostics, setting the again changed text
Plugin output: **an array of names of files that have been changed, or an empty array if none has changed**
migrate.ts input: **an array of names of files that have been changed, or an empty array if none has changed**
What migrate.ts does: save each file with their final text with the service

Reply:
After this PR:
**cli** - **validates user input**, pre-processes package.json and post-process **report / PR** 
↓
**migrate** - configures list of plugins, files to apply plugins and compiler options, then **run a service**
↓
**service** - compiles program and provides access to source files (cached files)... runs plugins on files (_to be done_) and saves updated files to a file system (sync) after all plugins applied to file.
↓
**plugin** - **receive filename to apply changes**, uses service to access file content... can also access to different files of create new ones, **returns the list of files been touched** (after this PR)

**Short:**
- cli - user input and output
- migrate - setup the project to compile
- service - access to source files in file system and logic to run plugins
- plugin - responsible to modify one or more files (cached)
- transform - part of the one plugin, responsible to modify AST

Seems legit :)